kahuna/public/js/util/async.js

```diff
@@ -79,6 +79,28 @@ async.factory("apiPoll", [
   }
 ]);
 
+async.factory("apiStatusPoll", [
+  "$q",
```

Comment:
This function isn't really needed as apiPoll takes a function which returns a promise and will continue to run it repeatedly until it the promise resolves.

Here's an example where it's used to do a similar check.
https://github.com/guardian/grid/blob/b1d75cd0bb349374d114f21b8e5aee2958e3e75f/kahuna/public/js/services/label.js#L52